src/Akka.Persistence.RavenDB/RavenDbPersistence.cs

```diff
@@ -1,74 +1,128 @@
         }
 
-        public static Config DefaultConfiguration()
-        {
-            return ConfigurationFactory.FromResource<RavenDbPersistence>("Akka.Persistence.RavenDb.reference.conf");
-        }
-
         private readonly CancellationTokenSource _stopTokenSource = new CancellationTokenSource();
 
         public void Stop()
         {
+            File.AppendAllText(@"C:\\Work\\Akka\\testLogs.txt",
+                $"\n\n(Stop cts) {Configuration.Name}:\n {Environment.StackTrace}");
             _stopTokenSource.Cancel();
-            //TODO stav: should dispose store here?
         }
 
         public CancellationTokenSource GetCancellationTokenSource(bool useSaveChangesTimeout)
         {
             var cts = CancellationTokenSource.CreateLinkedTokenSource(_stopTokenSource.Token);
-            if(useSaveChangesTimeout)
-                cts.CancelAfter(JournalConfiguration.SaveChangesTimeout);
+            if (useSaveChangesTimeout)
+                cts.CancelAfter(Configuration.SaveChangesTimeout);
             return cts;
         }
 
+        public async Task<Status> CreateDatabaseAsync()
+        {
+            using var cts = GetCancellationTokenSource(useSaveChangesTimeout: false);
+            var tries = 5;
+            while (tries > 0)
+            {
+                try
+                {
+                    var record =
+                        await Instance.Maintenance.Server.SendAsync(new GetDatabaseRecordOperation(Configuration.Name),
+                            token: cts.Token);
+                    if (record != null)
+                        return new Status.Success(NotUsed.Instance);
+
+                    var res = await Instance.Maintenance.Server.SendAsync(
+                        new CreateDatabaseOperation(new DatabaseRecord(Configuration.Name)), token: cts.Token);
+
+                    using (var context = JsonOperationContext.ShortTermSingleUse())
+                    {
+                        await Instance.GetRequestExecutor(Configuration.Name)
+                            .ExecuteAsync(new WaitForRaftIndexCommand(res.RaftCommandIndex), context, token: cts.Token);
+                    }
+
+                    return new Status.Success(NotUsed.Instance);
+                }
+                catch (ConcurrencyException e)
+                {
+                    if (e.Message.Contains("exists") == false)
+                    {
+                        File.AppendAllText(@"C:\\Work\\Akka\\testLogs.txt",
+                            $"\n\n(journal CreateDatabaseAsync) {Configuration.Name}:\n{e}");
+                        return new Status.Failure(e);
+                    }
+
+                    // The database already exists
+                    return new Status.Success(NotUsed.Instance);
+                }
+                catch (Exception e) when (e is OperationCanceledException || e is TaskCanceledException)
+                {
+                    return new Status.Success(NotUsed.Instance);
```

Comment:
why success?

Reply:
Usually it will be cancelled because Akka is done and wants to dispose the actor. If we treat it as a failure, the Receive func will terminate the actor before the other final Success messages left in the pipe are passed on. Some tests failed because they expected a "success" message. We may need to separate the cases between cancelled-because-call-timeout-is-up and a valid predictable cancellation